deepchem/feat/graph_data.py

```diff
@@ -204,6 +205,52 @@ def to_dgl_graph(self, self_loop: bool = False):
 
         return g
 
+    def numpy_to_torch(self):
+        """Convert numpy arrays to torch tensors. This may be useful when you are using PyTorch Geometric with GraphData objects.
+
+        Example
+        -------
+        >>> num_nodes, num_node_features = 5, 32
+        >>> num_edges, num_edge_features = 6, 32
+        >>> node_features = np.random.random_sample((num_nodes, num_node_features))
+        >>> edge_features = np.random.random_sample((num_edges, num_edge_features))
+        >>> edge_index = np.random.randint(0, num_nodes, (2, num_edges))
+        >>> graph_data = GraphData(node_features, edge_index, edge_features)
+        >>> graph_data = graph_data.numpy_to_torch()
+        >>> print(type(graph_data.node_features))
+        <class 'torch.Tensor'>
+        """
+        import copy
+
+        import torch
+        graph_copy = copy.deepcopy(self)
+
+        graph_copy.node_features = torch.from_numpy(self.node_features).float()
+        graph_copy.edge_index = torch.from_numpy(self.edge_index).long()
+        if self.edge_features is not None:
+            graph_copy.edge_features = torch.from_numpy(
+                self.edge_features).float()
+        else:
+            graph_copy.edge_features = None
+        if self.node_pos_features is not None:
+            graph_copy.node_pos_features = torch.from_numpy(
+                self.node_pos_features).float()
+        else:
+            graph_copy.node_pos_features = None
+
+        graph_copy.kwargs = {}
+        for key, value in self.kwargs.items():
+            if isinstance(value, np.ndarray):
+                value = torch.from_numpy(value)
+                graph_copy.kwargs[key] = value
+                setattr(graph_copy, key, value)
+
+        if isinstance(self, BatchGraphData):
```

Comment:
It's not good in a parent class to have a specialization down to the child class; breaks the object hierarchy. Instead, you want to overwrite `numpy_to_torch()` in `BatchGraphData` to make this change. You can use a call to `super` to avoid duplicate code so you can reuse the main implementation to do the heavy lifting